dotcom-rendering/src/web/components/ShowHideContainers.importable.tsx

```diff
@@ -0,0 +1,56 @@
+import { storage } from '@guardian/libs';
+import { useOnce } from '../lib/useOnce';
+
+type ContainerStates = { [id: string]: string };
+
+const getContainerStates = (): ContainerStates => {
+	const item = storage.local.get(`gu.prefs.container-states`);
+
+	if (!item) {
+		return {};
+	}
+
+	return item;
+};
+
+export const ShowHideContainers = () => {
+	const containerStates = getContainerStates();
+
+	const toggleContainer = (element: HTMLElement) => {
```

Comment:
Could this be defined outside of the component or use `useMemo` so that the function is re-defined on every render (in reality I think this will only render once but it's probably still best practice :))

Reply:
I can take a look at defining it outside, the reason it was inside is because it relies on the containerStates variable... I'm not entirely sure how an Island handles variables if they're defined outside of a component but I'll give it a go!